pkg/kubelet/kubelet.go

```diff
@@ -2289,32 +2290,74 @@ func (kl *Kubelet) rejectPod(pod *api.Pod, reason, message string) {
 		Message: "Pod " + message})
 }
 
+// getNodeAnyWay() must return a *api.Node which is required by RunGeneralPredicates().
+// The *api.Node is obtained as follows:
+// 1. When kubelet is in standaloneMode, manufactures *api.Node with kubelet
+// 2. Try to get *api.Node from kubelet's nodeInfo, if failed, we try 3.
+// 3. Try to get *api.Node from apiserver
+func (kl *Kubelet) getNodeAnyWay() (*api.Node, error) {
+	if kl.standaloneMode {
+		// I would prefer keeping the code below in comments, because:
+		// if kl.initialNodeStatus() returns with non-nil err, then the returned
+		// "node" is nil, which is not accepted for RunGeneralPredicates(). However,
+		// directly using `return kl.initialNodeStatus()` does not break any test for now.
+		// if n, err := kl.initialNodeStatus(); err == nil {
+		// 	return n, nil
+		// }
+		return kl.initialNodeStatus()
+	}
+	// if err != nil, n returns as nil which is not acceptable for RunGeneralPredicates()
```

Comment:
We should handle error before invoking `RunGeneralPredicates()`. The action might be that of rejecting a pod or failing kubelet itself.